src/graph/validator/MatchValidator.cpp

```diff
@@ -192,6 +192,12 @@ Status MatchValidator::buildNodeInfo(const MatchPath *path,
     auto alias = node->alias();
     auto *props = node->props();
     auto anonymous = false;
+    // if there exists some property with no tag, return a semantic error
+    if (node->labels() == nullptr && props != nullptr) {
```

Comment:
Directly `props != nullptr`?

Reply:
Seems ok to remove `node->labels() == nullptr`. Removed.